gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/query_service/api/ApiEventQueryServiceImplTest.java

```diff
@@ -0,0 +1,114 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@ExtendWith(MockitoExtension.class)
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+class ApiEventQueryServiceImplTest {
+
+    @Mock
+    EventLatestRepository eventLatestRepository;
+
+    @Captor
+    ArgumentCaptor<EventCriteria> eventCriteriaCaptor;
+
+    ApiEventQueryService cut;
+
+    @BeforeEach
+    void setUp() {
+        cut = new ApiEventQueryServiceImpl(eventLatestRepository);
+    }
+
+    @Test
+    void search_should_return_empty_if_no_event() {
+        when(eventLatestRepository.search(any(), eq(Event.EventProperties.API_ID), eq(0L), eq(1L))).thenReturn(List.of());
+        final Optional<Api> lastPublishedApi = cut.findLastPublishedApi("org-id", "env-id", "api-id");
+        assertThat(lastPublishedApi).isEmpty();
+        verifyEventCriteria();
+    }
+
+    @SneakyThrows
+    @Test
+    void search_should_return_last_published_api() {
+        final Api api = ApiFixtures.aProxyApiV4();
+
+        final Event event = new Event(
+            "event-id",
+            Set.of("env-id"),
+            EventType.PUBLISH_API,
+            GraviteeJacksonMapper.getInstance().writeValueAsString(ApiAdapter.INSTANCE.toRepository(api)),
+            null,
+            Map.of(),
+            new Date(),
+            new Date()
+        );
+        when(eventLatestRepository.search(any(), eq(Event.EventProperties.API_ID), eq(0L), eq(1L))).thenReturn(List.of(event));
+        final Optional<Api> lastPublishedApi = cut.findLastPublishedApi("org-id", "env-id", "api-id");
+        assertThat(lastPublishedApi)
+            .hasValueSatisfying(result -> {
+                assertThat(result.getApiDefinitionV4()).isEqualTo(api.getApiDefinitionV4());
+            });
+        verifyEventCriteria();
+    }
+
+    private void verifyEventCriteria() {
+        verify(eventLatestRepository).search(eventCriteriaCaptor.capture(), eq(Event.EventProperties.API_ID), eq(0L), eq(1L));
+        final EventCriteria builtEventCriteria = eventCriteriaCaptor.getValue();
+        assertThat(builtEventCriteria)
+            .satisfies(eventCriteria -> {
+                assertThat(eventCriteria.getEnvironments()).hasSize(1).containsExactly("env-id");
+                assertThat(eventCriteria.getTypes()).hasSize(1).containsExactly(EventType.PUBLISH_API);
+                assertThat(eventCriteria.getProperties())
+                    .hasSize(1)
+                    .containsExactly(Map.entry(Event.EventProperties.API_ID.getValue(), "api-id"));
```

Comment:
FYI, I think the `hasSize` assertions is redundant when you use `containsExactly`